### DISCUSSION TITLE
3.1

### DISCUSSION THREAD

**Discussion by EchterAlsFake** [EchterAlsFake/Porn_Fetch#25] [Announcements]
# 3.1

# Features / Improvements
- Added playlist support #20
- Added model for EPorner
- Error handling if wrong URL was entered
- Added Disclaimer to the License agreement
- Added more ui popups into the translations
- Updated **ALL** Stylesheets. The GUI now looks a LOT better
- implemented logic, so that user can't log in to PornHub without entering actual credentials
- Added searching support for EPorner
- Implemented update mechanism
- Implemented a factory reset for the settings
- Reimplemented back the client delay (optional)
- Redesigned settings widget
- Added get_by_category for EPorner
- Added a setting for specifying the workers and timeout in the threaded preset

# Bug Fixes / Issues
- Fixed an issue, where the settings were displayed, instead of the download tab
- Fixed total progress callback for processed videos in file mode
- Fixed the issue that the downloaded ffmpeg .zip wasn't deleted on windows
- Fixed callback progress for HQPorner
- Fixed the total progress for FFMPEG
- Fixed the FFMPEG type error #21
- Fixed individual progress for FFMPEG
- Fixed searching (result) limit
- Fixed total progress for loading objects and stuff
- Handling `Client.call failed after 4 attempts`
- Fixed an issue that video reversing is not working
- Fixed overflow error
- Progress now gets only updated all 0.5 seconds
- Some issues in the CLI

# Code related
- Renamed UI widgets so that they make more sense
- Improved the load style function
- Added code comments
- Code refactoring (refactored all Signals)

# Android
- Fixed the threaded presets for Android (if you still experience issues, use less workers in settings)

# Downloads:

### Android is ONLY for arm-v8a architecture!

<hr /><em>This discussion was created from the release <a href='https://github.com/EchterAlsFake/Porn_Fetch/releases/tag/3.1'>3.1</a>.</em>

**Comment by xero9826** [comment]
Hi there, there's no way to modify or create a copy of none metadata file.mp4, can't organizes videos without it.

**Reply by EchterAlsFake** [reply]
So you want me to add a feature which lets you edit the metadata of the file to specify the title, author and release date of it. Did I understand that right?
